source/agora/network/Manager.d

```diff
@@ -463,6 +464,7 @@ public class NetworkManager
         this.discovery_task = new AddressDiscoveryTask(&this.addAddresses);
         this.clock = clock;
         this.owner_node = owner_node;
+        this.resolver = resolver;
```

Comment:
Ok, this is a decision point and may be conflicting with #2703 's caching registry.
I think (we might also rename it but) we should decide tasks of the `DNSResolver`. Currently it asks queries to configured peers and needs an implementation to cache and manage TTLs (which the mentioned PR implements it on `Registry since we already have a storage there). After the mentioned PR we can have all nodes to have registry with type `caching` enabled default, thus requesting a `getValidator` will eventually pass through its own registry.

Reply:
Well, we could just add the local caching registry to `resolv.conf` and utilize it that way.
NetworkClient will have to handle expiration of TTL. But I beleive those are not within the scope of this PR

Reply:
Yeah, #2703 already implements a caching registry with TTL support.
^ With that, then this PR also shouldn't include a resolver in the Manager, might be too early for mentioned reasons.